Tests/GRPCNIOTransportHTTP2Tests/HTTP2TransportTLSEnabledTests.swift

```diff
@@ -89,52 +88,58 @@ struct HTTP2TransportTLSEnabledTests {
-    arguments: [TransportKind.posix],
-    [TransportKind.posix]
+    arguments: [TransportKind.posix, .transportServices],
+    [TransportKind.posix, .transportServices]
   )
   // Verification should fail because the custom hostname is missing on the client.
   func testClientFailsServerValidation(
     clientTransport: TransportKind,
     serverTransport: TransportKind
   ) async throws {
     let certificateKeyPairs = try SelfSignedCertificateKeyPairs()
-    let clientConfig = self.makeMTLSClientConfig(
+    let clientTransportConfig = self.makeDefaultTLSClientConfig(
       for: clientTransport,
       certificateKeyPairs: certificateKeyPairs,
-      serverHostname: "the-wrong-hostname"
+      authority: nil
     )
-
-    let serverConfig = self.makeMTLSServerConfig(
+    let serverTransportConfig = self.makeDefaultTLSServerConfig(
       for: serverTransport,
-      certificateKeyPairs: certificateKeyPairs,
-      includeClientCertificateInTrustRoots: true
+      certificateKeyPairs: certificateKeyPairs
     )
 
     try await self.withClientAndServer(
-      clientConfig: clientConfig,
-      serverConfig: serverConfig
+      clientConfig: clientTransportConfig,
+      serverConfig: serverTransportConfig
     ) { control in
       await #expect {
         try await self.executeUnaryRPC(control: control)
       } throws: { error in
-        guard let rootError = error as? RPCError else {
-          Issue.record("Should be an RPC error")
-          return false
-        }
+        let rootError = try #require(error as? RPCError)
         #expect(rootError.code == .unavailable)
-        #expect(
-          rootError.message
-            == "The server accepted the TCP connection but closed the connection before completing the HTTP/2 connection preface."
-        )
 
-        guard
-          let sslError = rootError.cause as? NIOSSLExtraError,
-          case .failedToValidateHostname = sslError
-        else {
-          Issue.record(
-            "Should be a NIOSSLExtraError.failedToValidateHostname error, but was: \(String(describing: rootError.cause))"
+        switch clientTransport {
+        case .posix:
+          #expect(
+            rootError.message
+              == "The server accepted the TCP connection but closed the connection before completing the HTTP/2 connection preface."
           )
-          return false
+          let sslError = try #require(rootError.cause as? NIOSSLExtraError)
+          guard sslError == .failedToValidateHostname else {
+            Issue.record(
+              "Should be a NIOSSLExtraError.failedToValidateHostname error, but was: \(String(describing: rootError.cause))"
+            )
+            return false
+          }
+
+        case .transportServices:
+          #expect(rootError.message.starts(with: "Could not establish a connection to"))
+          let nwError = try #require(rootError.cause as? NWError)
+          guard case .tls(-9808) /* bad certificate format */ = nwError else {
```

Comment:
There are constants for error codes, this one is https://developer.apple.com/documentation/security/errsslbadcert

https://www.osstatus.com is pretty useful way to find the name from a value.